criu/vdso-compat.c

```diff
@@ -13,6 +13,9 @@ static void exit_on(int ret, int err_fd, char *reason)
 	if (ret) {
 		syscall(__NR_write, err_fd, reason, strlen(reason));
 		syscall(__NR_exit, ret);
+#ifdef __COVERITY__
+		__coverity_panic__();
+#endif
```

Comment:
Can we use `__noreturn` in the function definition instead?

Reply:
I just tried to add `__attribute__((__noreturn__))` and gcc says:
```
  CC       criu/vdso-compat.o
criu/vdso-compat.c: In function ‘exit_on’:
criu/vdso-compat.c:22:1: error: ‘noreturn’ function does return [-Werror]
   22 | }
      | ^
cc1: all warnings being treated as errors
```

Any other ways to use `__noreturn__`?

Reply:
Ah, yeah, I've missed that it returns on `(!ret)`, sorry.
Maybe like in kenel `unreachable()` rather than this coverity iffdeffery, what do you think?

Reply:
Using `__builtin_unreachable();` seems to work.